modules/engine-prototype/src/main/scala/engine/Step.scala

```diff
@@ -0,0 +1,251 @@
+package engine
+
+import scala.concurrent.ExecutionContext
+
+import cats.implicits._
+import cats.effect.Effect
+
+sealed trait Step
+object Step {
+
+  // Whatever is shared among all Steps. Notice it doesn't // extend Step
+  // because it's meant to be used only internally
+  case class Core(
+    id: String,
+    config: Core.Configuration,
+    breakpoint: Boolean
+  )
```

Comment:
This should have been private to the `Step` object but I had some issues with scoping that I don't remember exactly. This contains all the shared information of all the Steps. All Steps have Core Step as a composite value.

Now that I'm writing this I realize `breakpoint` shouldn't go here, it only makes sense in the Pending Steps.